Mathlib/RingTheory/Polynomial/Wronskian.lean

```diff
@@ -0,0 +1,114 @@
+
+@[simp]
+theorem wronskian_zero_left (a : R[X]) : wronskian 0 a = 0 := by
+  rw [← wronskianBilin_apply 0 a, map_zero]; rfl
+
+@[simp]
+theorem wronskian_zero_right (a : R[X]) : wronskian a 0 = 0 := (wronskianBilin R a).map_zero
+
+theorem wronskian_neg_left (a b : R[X]) : wronskian (-a) b = -wronskian a b :=
+  LinearMap.map_neg₂ (wronskianBilin R) a b
+
+theorem wronskian_neg_right (a b : R[X]) : wronskian a (-b) = -wronskian a b :=
+  (wronskianBilin R a).map_neg b
+
+theorem wronskian_add_right (a b c : R[X]) : wronskian a (b + c) = wronskian a b + wronskian a c :=
+  (wronskianBilin R a).map_add b c
+
+theorem wronskian_add_left (a b c : R[X]) : wronskian (a + b) c = wronskian a c + wronskian b c :=
+  (wronskianBilin R).map_add₂ a b c
+
+theorem wronskian_self_eq_zero (a : R[X]) : wronskian a a = 0 := by
+  rw [wronskian, mul_comm, sub_self]
+
+theorem isAlt_wronskianBilin : (wronskianBilin R).IsAlt := wronskian_self_eq_zero
+
+theorem wronskian_neg_eq (a b : R[X]) : -wronskian a b = wronskian b a :=
+  LinearMap.IsAlt.neg isAlt_wronskianBilin a b
+
+theorem wronskian_eq_of_sum_zero {a b c : R[X]} (hAdd : a + b + c = 0) :
+    wronskian a b = wronskian b c := isAlt_wronskianBilin.eq_of_add_add_eq_zero hAdd
+
+private theorem degree_ne_bot {a : R[X]} (ha : a ≠ 0) : a.degree ≠ ⊥ := by
+  intro h; rw [Polynomial.degree_eq_bot] at h; exact ha h
+
+theorem degree_wronskian_lt_add {a b : R[X]} (ha : a ≠ 0) (hb : b ≠ 0) :
+    (wronskian a b).degree < a.degree + b.degree := by
+  calc
+    (wronskian a b).degree ≤ max (a * derivative b).degree (derivative a * b).degree :=
+      Polynomial.degree_sub_le _ _
+    _ < a.degree + b.degree := by
+      rw [max_lt_iff]
+      constructor
+      case left =>
+        apply lt_of_le_of_lt
+        exact degree_mul_le a (derivative b)
+        rw [WithBot.add_lt_add_iff_left (degree_ne_bot ha)]
+        exact Polynomial.degree_derivative_lt hb
+      case right =>
+        apply lt_of_le_of_lt
+        exact degree_mul_le (derivative a) b
+        rw [WithBot.add_lt_add_iff_right (degree_ne_bot hb)]
+        exact Polynomial.degree_derivative_lt ha
+
+theorem natDegree_wronskian_lt_add {a b : R[X]} (hw : wronskian a b ≠ 0) :
+    (wronskian a b).natDegree < a.natDegree + b.natDegree := by
+  have ha : a ≠ 0 := by intro h; subst h; rw [wronskian_zero_left] at hw; exact hw rfl
+  have hb : b ≠ 0 := by intro h; subst h; rw [wronskian_zero_right] at hw; exact hw rfl
+  rw [← WithBot.coe_lt_coe, WithBot.coe_add]
+  convert ← degree_wronskian_lt_add ha hb
+  · exact Polynomial.degree_eq_natDegree hw
+  · exact Polynomial.degree_eq_natDegree ha
+  · exact Polynomial.degree_eq_natDegree hb
+
+-- Note: the following is false!
+-- Counterexample: b = a = 1 →
+-- (wronskian a b).natDegree = a.natDegree = b.natDegree = 0
+/-
+lemma wronskian.natDegree_lt_add {a b : R[X]}
+  (ha : a ≠ 0) (hb : b ≠ 0) :
+  (wronskian a b).natDegree < a.natDegree + b.natDegree := sorry
+-/
```

Comment:
I think this would be better placed in the docstring of `natDegree_wronskian_lt_add`, something like

> Note this would be false with just `(ha : a ≠ 0) (hb : b ≠ 0)`, as when `a = b = 1` we have `(wronskian a b).natDegree = a.natDegree = b.natDegree = 0`.